### ISSUE TITLE
null as a result of http.get

### ISSUE THREAD

**Issue by smimram** [savonet/liquidsoap#1688]
I am not sure that having a nullable value as result of `http.get` is such a great idea.

First, the script
```
c = http.get("https://raw.github.com/savonet/liquidsoap/master/CHANGES.md")
print("Here are the latest changes in Liquidsoap:\n\n" ^ c)
```
is failing because we are concatenating with a nullable value.

Second, the doc is hard to read
```
Type: (?headers : [string * string], ?http_version : string?,
 ?redirect : bool, ?timeout : int, string) -> string
.{
  headers : [string * string],
  status_message : string,
  status_code : int,
  protocol_version : string}?
```
compared to what it was before:
```
(?headers : [string * string], ?http_version : string?,
 ?redirect : bool, ?timeout : int, string) -> string
```
(without the nullable, methods are showns in their own section).

Since this is only for timeout, we could easily come up with a fake error sur as 524 which is apparently used by cloudflare for timeouts.

**Comment by toots** [comment]
I understand why what other mechanism do you propose to account for timeouts when executing the query? I can think of the following:
1. Return `999` response code as before. Very hacky, we should be able to do better
2. Raise an exception. Nothing needed from the user but it'll blow-up in their face even perhaps after testing the script and putting it in production
3. Return `null` and force them to account for this case while working on their script.

Option 3 sounds the best to me..

**Comment by toots** [comment]
Ha I didn't see your suggestion for `524`. Hmm if couldflare does it, why not.

**Comment by toots** [comment]
Closed with 0ba2ac35
